docs: Add missing duplex property to RequestInit and update fetch()

Shrinivassab opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. 🙌

Add details below to help us review your pull request (PR).
Explain your changes and link to a related issue or pull request.
Your PR may be delayed or closed if you don't provide enough information. -->

### Description

<!-- ✍️ Summarize your changes in one or two sentences. -->
- Adds missing `duplex` property to `RequestInit` (alphabetically between `credentials` and `headers`).
- Documents `TypeError` behavior in `fetch()` exceptions table when:
  - `body` is a `ReadableStream` but `duplex` is omitted/not `"half"`.
  - Invalid `duplex` values are provided.

### Motivation

<!-- ❓ Why are you making these changes and how do they help readers? -->

### Additional details

<!-- 🔗 Link to release notes, browser docs, bug trackers, source control, or other resources. -->

### Related issues and pull requests

Fixes #40086

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request must be merged first, use "**Depends on:** #123" -->

<!-- 🔎 After submitting, the 'Checks' tab of your PR shows the build status. -->

Shrinivassab committed:
docs: Add missing duplex property to RequestInit and update fetch() exceptions

Shrinivassab committed:
docs: Add missing duplex property to RequestInit and update fetch() exceptions

wbamberg commented on files/en-us/web/api/window/fetch/index.md:
```diff
@@ -175,6 +175,22 @@ fetch("https://example.com/", {
+      </td>
+      <td>
+        <pre class="brush: js">
+          // Missing or invalid duplex with ReadableStream
```
```suggestion
```

None of the other examples here have comments like this.

wbamberg commented on files/en-us/web/api/window/fetch/index.md:
```diff
@@ -175,6 +175,22 @@ fetch("https://example.com/", {
+          fetch("https://example.com", {
+            method: "POST",
+            body: stream,
+            // Missing: duplex: "half"
```
```suggestion
```

wbamberg reviewed (commented):
Thank you @Shrinivassab ! I realised we also need a page for the corresponding `duplex` property of `Request`. You could model this on, for example, https://developer.mozilla.org/en-US/docs/Web/API/Request/redirect.

Shrinivassab committed:
Merge branch 'main' into add-duplex-docs

Shrinivassab committed:
Update files/en-us/web/api/requestinit/index.md

Accepted PR commented...

Co-authored-by: wbamberg <will@bootbonnet.ca>

Shrinivassab committed:
Update files/en-us/web/api/requestinit/index.md

Accepted PR commented...

Co-authored-by: wbamberg <will@bootbonnet.ca>

Shrinivassab committed:
Update files/en-us/web/api/window/fetch/index.md

Accepted PR commented...

Co-authored-by: wbamberg <will@bootbonnet.ca>

Shrinivassab committed:
Update files/en-us/web/api/window/fetch/index.md

Accepted PR commented...

Co-authored-by: wbamberg <will@bootbonnet.ca>

Shrinivassab committed:
Update files/en-us/web/api/window/fetch/index.md

Accepted PR commented...

Co-authored-by: wbamberg <will@bootbonnet.ca>

Shrinivassab committed:
Update files/en-us/web/api/window/fetch/index.md

Accepted PR commented...

Co-authored-by: wbamberg <will@bootbonnet.ca>

Shrinivassab committed:
Merge branch 'add-duplex-docs' of https://github.com/Shrinivassab/content into add-duplex-docs

Shrinivassab committed:
docs(Request): Add duplex property documentation

Shrinivassab committed:
docs(Request): Add duplex property documentation

Shrinivassab commented:
Hi @wbamberg I have incorporated your PR comments and in addition to that i added new page for duplex: Please go over it and share your review comments. I will update it based on your suggestions. 
--- 
![localhost_5042-Request duplex property - Web APIs  MDN-fpscreenshot](https://github.com/user-attachments/assets/b0dc72ac-7628-4235-82ca-16d81973db65)

wbamberg commented on files/en-us/web/api/request/duplex/index.md:
```diff
@@ -0,0 +1,67 @@
+
+A string with the following possible value:
+
+- `"half"`: The browser must send the entire request before processing the response. This is required when the request body is a {{domxref("ReadableStream")}}.
```
```suggestion
- `"half"`
  - : The browser must send the entire request before processing the response.
```

("this is required" is about the options argument, not the property.)

wbamberg committed:
Update files/en-us/web/api/request/duplex/index.md

wbamberg committed:
Update files/en-us/web/api/request/duplex/index.md

wbamberg commented on files/en-us/web/api/request/duplex/index.md:
````diff
@@ -0,0 +1,68 @@
+
+console.log(request.duplex); // "half"
+```
+
+### Handling missing duplex with ReadableStream
+
+```js
+try {
+  // This will throw if duplex is missing with ReadableStream
+  const invalidRequest = new Request("/upload", {
+    method: "POST",
+    body: new ReadableStream(),
+  });
+} catch (err) {
+  console.error(err); // TypeError: Failed to construct 'Request'
+}
+```
````
```suggestion
```

This example is not needed.

wbamberg committed:
Update files/en-us/web/api/request/duplex/index.md

wbamberg committed:
Update files/en-us/web/api/request/duplex/index.md

wbamberg commented on files/en-us/web/api/request/duplex/index.md:
```diff
@@ -0,0 +1,54 @@
+- `"half"`
+  - : The browser must send the entire request before processing the response.
+
+If not specified when creating the request, the value will be `undefined`.
```
This seems to be incorrect:

<img width="313" alt="Screenshot 2025-06-30 at 3 41 50 PM" src="https://github.com/user-attachments/assets/b8b5383d-f785-4872-86f1-4ceb5d8f4c49" />

...and it would be weird if it were correct, because what would that mean as a value?

Shrinivassab committed:
Merge branch 'main' into add-duplex-docs

Shrinivassab committed:
Merge branch 'add-duplex-docs' of https://github.com/Shrinivassab/content into add-duplex-docs

caugner committed:
Merge branch 'main' into add-duplex-docs

Shrinivassab committed:
Merge branch 'main' into add-duplex-docs

Shrinivassab committed:
Removed undefined and added warning note

Josh-Cena committed:
Update files/en-us/web/api/window/fetch/index.md

wbamberg reviewed (approved):
👍 thank you @Shrinivassab , and sorry to be slow reviewing.
